### ISSUE TITLE
[Accessibility] Focus jumps to page with submission error

### ISSUE THREAD

**Issue by tallulahkay** [civiform/civiform#11924]
**Describe the bug**
When a user submits a form without selecting required locations, an error message appears, but the error is not announced to screen reader users because focus jumps to the top of the page instead of to the error message.

**WCAG Success Criterion**
[3.3.1 Error Identification (Level A)](https://www.w3.org/TR/WCAG21/#error-identification)
[3.3.3 Error Suggestion (Level AA)](https://www.w3.org/TR/WCAG21/#error-suggestion)

**To Reproduce**
1. Enable screen reader (VoiceOver, NVDA, JAWS, TalkBack)
2. Navigate to a location selection question marked as required
3. Do not select any locations
4. Press the submit/continue button
5. Observe that error appears but is not announced

**Expected behavior**
When form submission fails due to validation errors:
- The error message should be announced immediately to screen reader users
- Focus should move to the first location checkbox
